packages/clarity-js/src/layout/dom.ts

```diff
@@ -115,10 +116,12 @@ export function add(node: Node, parent: Node, data: NodeInfo, source: Source): v
         selector: null,
         hash: null,
         region: regionId,
-        metadata: { active: true, suspend: false, privacy, position: null, size: null }
+        metadata: { active: true, suspend: false, privacy, position: null, size: null },
+        fragment: fragmentId,
     };
 
     updateSelector(values[id]);
+    if (values[id].fragment && updatedFragments.indexOf(values[id].fragment) === -1) { updatedFragments.push(values[id].fragment); }
```

Comment:
>if (values[id].fragment && updatedFragments.indexOf( [](http://example.com/codeflow?start=4&length=52)

Can you do this as part of "track" function - since it's purpose is to see if we should track a node for update; we could track fragment there too. It's also shared between add & update - therefore less code duplication.